CriteoPublisherSdk/Tests/UnitTests/Standalone/CRBannerViewTests.m

```diff
@@ -32,123 +32,75 @@
 #import "NSURL+Criteo.h"
 #import "CR_TokenValue+Testing.h"
 #import "XCTestCase+Criteo.h"
+#import "CR_DependencyProvider.h"
+#import "CR_DependencyProvider+Testing.h"
 
 @import WebKit;
 
 @interface CRBannerViewTests : XCTestCase <WKNavigationDelegate>
 
 @property(nonatomic, copy) void (^webViewDidLoadBlock)(void);
-@property(nonatomic, strong) CR_CacheAdUnit *cacheAdUnit;
+@property(nonatomic, strong) CR_CacheAdUnit *expectedCacheAdUnit;
 @property(nonatomic, strong) CRBannerAdUnit *adUnit;
 @property(strong, nonatomic) CR_URLOpenerMock *urlOpener;
+@property(strong, nonatomic) Criteo *criteo;
 
 @end
 
 @implementation CRBannerViewTests
 
 - (void)setUp {
-  _cacheAdUnit = nil;
-  _adUnit = nil;
+  self.expectedCacheAdUnit = [[CR_CacheAdUnit alloc] initWithAdUnitId:@"123"
+                                                                 size:CGSizeMake(47.0, 57.0)
+                                                           adUnitType:CRAdUnitTypeBanner];
+  self.adUnit = [[CRBannerAdUnit alloc] initWithAdUnitId:@"123" size:CGSizeMake(47.0, 57.0)];
   self.urlOpener = [[CR_URLOpenerMock alloc] init];
-}
 
-- (CR_CacheAdUnit *)expectedCacheAdUnit {
-  if (!_cacheAdUnit) {
-    _cacheAdUnit = [[CR_CacheAdUnit alloc] initWithAdUnitId:@"123"
-                                                       size:CGSizeMake(47.0, 57.0)
-                                                 adUnitType:CRAdUnitTypeBanner];
-  }
-  return _cacheAdUnit;
+  CR_DependencyProvider *dependencyProvider = CR_DependencyProvider.testing_dependencyProvider;
+  self.criteo = OCMPartialMock([Criteo.alloc initWithDependencyProvider:dependencyProvider]);
 }
 
-- (CRBannerAdUnit *)adUnit {
-  if (!_adUnit) {
-    _adUnit = [[CRBannerAdUnit alloc] initWithAdUnitId:@"123" size:CGSizeMake(47.0, 57.0)];
-  }
-  return _adUnit;
+- (void)tearDown {
+  self.criteo = nil;
```

Comment:
Is this really required?

Reply:
Long story short: another painful testRendering (remember the ultra flaky one ??)

I do not know why but there is a hidden side effect somewhere that break this test even if the body of this test remain unchanged. Setting criteo to nil (although it is setup to new instance every time) seems to fix the test.

I tried to do the same on the testRendering from interstitial with no chance :/

Reply:
I think that would deserve some comment

Reply:
On second thoughts, that does not makes sense as this property should be deallocated anyway, TestCases are allocated at each test run, and therefore dealloc is implicit after. Are you sure this has effects?

Reply:
Now I remember we has seen similar things in the past with Romain, will ask him, I suspect there is a bug or some kind of subtlety in [OCMock partial mocks](https://ocmock.org/reference/#partial-mocks) they state that it should just works, but it apparently does not.

Reply:
![image](https://user-images.githubusercontent.com/3816099/88656119-19a15580-d0d0-11ea-983e-82b0912a3c10.png)

I will add a comment

Reply:
done